### ISSUE TITLE
Error compiling iterator_ops.cc when building from source with MSVC 2017

### ISSUE THREAD

**Issue by RugnirViking** [tensorflow/tensorflow#24046]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10 
- TensorFlow installed from (source or binary): Source
- TensorFlow version: latest from git - 1.12.0
- Python version: 2.7
- Bazel version (if compiling from source): 0.19.2
- GCC/Compiler version (if compiling from source): 
- CUDA/cuDNN version: N/A
- GPU model and memory: NVIDIA GeForce 940MX 4096mb

**When building tensorflow from source using bazel on windows 10 I encountered an error in iterator_ops.cc.**
To the best of my ability I think the error that stopped the build was 
`"binary '*': no operator found which takes a left-hand operand of type 'const tensorflow::OpInputList::Iterator' (or there is no acceptable conversion)"`
But I'm not familliar with tensorflow or the building process being used.

**Provide the exact sequence of commands / steps that you executed before running into the problem**
bazel build :loader (loader being a barebones .cc file with tensorflow as a dependancy)

**Any other info / logs**
The relevant build logs: (somewhat long error message)
https://pastebin.com/akqRxB0P

**Comment by ymodak** [comment]
TensorFlow is tested against the Visual Studio 2015 Update 3. Can you please install Install the Visual Studio 2015 and try again?

**Comment by RugnirViking** [comment]
I have done so and still recieved an error find the logs here
It appears to be a different issue, but its hard for me to be sure. These logs are pretty cryptic :)
https://pastebin.com/mFq0mTwH

**Comment by ymodak** [comment]
Python 3.7 is not officially supported version for TF 1.12. Adding gunan.

**Comment by RugnirViking** [comment]
I'll give it a go with Python 3.6 and see how that goes. It takes over an hour so I'll have to get back to you

**Comment by RugnirViking** [comment]
I have completed the attempt with python 3.6. Still no luck - failed again right at the end of the build.
This is still following the tensorflow guidance now from: https://www.tensorflow.org/guide/extend/cc

The logs of the error with python 3.6 can be found here:
https://pastebin.com/14fsGd5k

**Comment by RugnirViking** [comment]
@gunan Sorry to ping you- I hate to be a nag. I've been blocked by this issue for several days now. It would be useful to get an ETA of when you can take a look at this.

**Comment by gunan** [comment]
Sorry for missing this issue before.
What is the reason for building from sources? Could you try installing from prebuilt binaries?
For CPU version, it should be very straightforward.

**Comment by RugnirViking** [comment]
At the moment I have resorted to piping the data over a network to a python server to get predictions that way. It would be far nicer to have it working in native c++ however.

The model is already trained and in a h5 file (keras) and a .pb (tensorflow) file - these work well already. I had just hoped to be able to import one of these using native cpp and make predictions using it. I had tried to follow any tutorial I had found for the last week or so.
